Mathlib/SetTheory/Ordinal/Nimber.lean

```diff
@@ -356,4 +369,25 @@ theorem add_trichotomy {a b c : Nimber} (h : a + b + c ≠ 0) :
   · rw [← hx'] at hx
     exact Or.inr <| Or.inr hx
 
+/-- Nimber addition of naturals corresponds to the bitwise XOR operation. -/
+theorem add_nat (a b : ℕ) : ∗a + ∗b = ∗(a ^^^ b) := by
+  apply le_antisymm
+  · apply add_le_of_forall_ne
+    all_goals
+      intro c hc
+      obtain ⟨c, rfl⟩ := eq_nat_of_le_nat hc.le
+      rw [OrderIso.lt_iff_lt] at hc
+      replace hc := Nat.cast_lt.1 hc
```

Comment:
```suggestion
      rw [OrderIso.lt_iff_lt, Nat.cast_lt] at hc
```
seems surely better (you might need a \l on the second one, I didn't check)

Reply:
For whatever reason, the termination checker fails when doing this. I'm not really sure what's going on there.

Reply:
Ah, probably the old version is kept around secretly somewhere, and the termination checker is using that as proof of termination